src/Core.tsx

```diff
@@ -199,7 +199,7 @@ export class Core extends React.Component<IModalProps, IModalState> {
   }
 
   // Interacts with RLogin to expose hide/show modal functions to the steps listed below
-  public showModalWithStep (step: 'Step1' | 'wrongNetwork' | 'walletInfo') {
+  public showModalWithStep (step: 'Step1' | 'changeNetwork' | 'walletInfo') {
```

Comment:
I finally discovered that we were missing this step in the list. This will show the same UI of `wrongNetwork` but with different text. See https://github.com/rsksmart/rLogin/blob/4795adb9204fbe34c82f4b05ab0301242959dec1/src/ux/wrongNetwork/WrongNetworkComponent.tsx#L41-L48 and https://github.com/rsksmart/rLogin/blob/4795adb9204fbe34c82f4b05ab0301242959dec1/src/Core.tsx#L537

Now this shows

<img width="477" alt="Screen Shot 2022-05-24 at 01 11 24" src="https://user-images.githubusercontent.com/36084092/169949016-600ce0d9-d464-4829-a92b-29337fa3e15e.png">

Reply:
This new language makes more sense but the functionality isn't implemented. So if the user is using Metamask then they will see that they can automatically connect to RSK Mainnet. But it also shows that the app can change to Ethereum but there is no way for the user to do so. They have to do it from their Provider.

If we want to include this functionality, then we should also implement the Metamask specific`wallet_addEthereumChain` method for the other networks. In that case, in #274 we should combine the information from [the adapters file](https://github.com/rsksmart/rLogin/pull/274/files#diff-5434d7621041e2bbe017e128e361d977a1e34958ea9a3ed45a2ea3e4450b9820R15-R58) with the [networks variable](https://github.com/rsksmart/rLogin/blob/develop/src/ux/wrongNetwork/changeNetwork.ts#L14-L43). There is a lot of duplication there and by single sourcing it developers can add additional networks quicker. 

That `networks` variable also includes a link to the blockexplorer which could be the default and then overwritten by the developer's option?

Reply:
I think we are talking about two different topics and we could open two discussion issues for this and keep this implementation for now.

Rationale:

### About _change network_ feature

This was created some time ago, and the idea was to:
1. Reuse the things we had built `wrongNetwork`
2. Open the door to implement new UX for changing network in other providers

I went back in the history, and for some reason `rLogin.showChangeNetwork` changed to show the hardware wallet UX `step = chooseNetwork` instead of `step = changeNetwork`

- PR adding this feature https://github.com/rsksmart/rLogin/pull/177
- How it was before changing it https://github.com/rsksmart/rLogin/blame/b38aae5965929fac33dc8661a88555d694fcf356/src/RLogin.tsx#L130
- Last change https://github.com/rsksmart/rLogin/blame/26c4994a75ed7e791e632e914e05dc8b5da8ceb1/src/RLogin.tsx#L103

I think that:
- we can keep this door open and use what we have, keeping the change network button and ask for feedback for that UX
- start thinking of a better UX for the other wallets
- (see below) understand what to do with the other networks in MetaMask

### About what to do with MetaMask `wallet_addEthereumChain`

See this related comment https://github.com/rsksmart/rLogin/issues/268#issuecomment-1135384736

I know this is not directly what we are discussing here but I bring up similar concerns. I would like to remark some:
- The idea of rLogin is to make easy to build dapps on RSK, that is why we started by implementing just that two networks. Some dapps are cross/multi-chain (like token bridge) so we started exploring on different networks while keeping "a generic UX" for them. Now I see a big change: we are moving away from generic UX into something more custom, we need to be careful 
- We want to keep the tool decentralized thus allow the developer to choose whatever node they want their users to use. A clear example of this is. See for example https://moralis.io/how-to-connect-to-polygon-nodes/
- We know insights about how RSK infra works, if we want to add refs to infra of other networks we will need to do further research. See for example https://docs.polygon.technology/docs/develop/network-details/network/
  
  > Public RPCs may have traffic or rate-limits depending on usage. You can sign up for a dedicated free RPC URL at the following
  
  This can become a bummer "I connected to Polygon with rLogin and I don't see my balance updated"

Reply:
Finally, we took a decision: we are not adding "change network" button to info screen now until we have a better UX for that. In consequence I will open issues for the things we discussed here.